builtin/providers/aws/resource_aws_api_gateway_domain_name_test.go

```diff
@@ -0,0 +1,206 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/apigateway"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAWSAPIGatewayDomainName_basic(t *testing.T) {
+	var conf apigateway.DomainName
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSAPIGatewayDomainNameDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccAWSAPIGatewayDomainNameConfigCreate(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSAPIGatewayDomainNameExists("aws_api_gateway_domain_name.test", &conf),
+					resource.TestCheckResourceAttr(
+						"aws_api_gateway_domain_name.test", "certificate_body", testAccAWSAPIGatewayCertBody,
+					),
+					resource.TestCheckResourceAttr(
+						"aws_api_gateway_domain_name.test", "certificate_chain", testAccAWSAPIGatewayCertChain,
+					),
+					resource.TestCheckResourceAttr(
+						"aws_api_gateway_domain_name.test", "certificate_name", "tf-acc-apigateway-domain-name",
+					),
+					resource.TestCheckResourceAttr(
+						"aws_api_gateway_domain_name.test", "certificate_private_key", testAccAWSAPIGatewayCertPrivateKey,
+					),
+					resource.TestCheckResourceAttr(
+						"aws_api_gateway_domain_name.test", "domain_name", "test-api-gateway-domain-cert0.com",
+					),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAWSAPIGatewayDomainNameExists(n string, res *apigateway.DomainName) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No API Gateway DomainName ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).apigateway
+
+		req := &apigateway.GetDomainNameInput{
+			DomainName: aws.String(rs.Primary.ID),
+		}
+		describe, err := conn.GetDomainName(req)
+		if err != nil {
+			return err
+		}
+
+		if *describe.DomainName != rs.Primary.ID {
+			return fmt.Errorf("APIGateway DomainName not found")
+		}
+
+		*res = *describe
+
+		return nil
+	}
+}
+
+func testAccCheckAWSAPIGatewayDomainNameDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).apigateway
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_api_gateway_api_key" {
+			continue
+		}
+
+		describe, err := conn.GetDomainNames(&apigateway.GetDomainNamesInput{})
+
+		if err == nil {
+			if len(describe.Items) != 0 &&
+				*describe.Items[0].DomainName == rs.Primary.ID {
+				return fmt.Errorf("API Gateway DomainName still exists")
+			}
+		}
+
+		aws2err, ok := err.(awserr.Error)
+		if !ok {
+			return err
+		}
+		if aws2err.Code() != "NotFoundException" {
+			return err
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+// Expires on July 18, 2026
```

Comment:
Yep, I think we can avoid all this complexity and static cert is fine here.
How about using wildcard cert and randomise the subdomain?